zypp/target/rpm/RpmDb.cc

```diff
@@ -331,6 +334,14 @@ void RpmDb::initDatabase( Pathname root_r, Pathname dbPath_r, bool doRebuild_r )
     ZYPP_THROW(RpmInvalidRootException(root_r, dbPath_r));
   }
 
+  if ( dbPath_r == "/var/lib/rpm"
+    && ! PathInfo( root_r/"/var/lib/rpm" ).isExist()
```

Comment:
Could use root_r / dbPath_r

Reply:
Yes. I just wanted to strengthen the fact that the code is not intended to be used with arbitrary dbPath_r.